commit 0f3d47199b95098b04ee774eb2730c2812bf4dd0
Author: Norman Maurer <norman_maurer@apple.com>
Date:   Thu Feb 4 13:41:02 2016 +0100

    [#4834] Fix race in AddressResolverGroup
    
    Motivation:
    
    We miss to use synchronized when remove the executor from the map.
    
    Modifications:
    
    Add synchronized(...) keyword
    
    Result:
    
    No more race.

diff --git a/resolver/src/main/java/io/netty/resolver/AddressResolverGroup.java b/resolver/src/main/java/io/netty/resolver/AddressResolverGroup.java
index 89b00df..509475e 100644
--- a/resolver/src/main/java/io/netty/resolver/AddressResolverGroup.java
+++ b/resolver/src/main/java/io/netty/resolver/AddressResolverGroup.java
@@ -73,7 +73,9 @@ public abstract class AddressResolverGroup<T extends SocketAddress> implements C
                 executor.terminationFuture().addListener(new FutureListener<Object>() {
                     @Override
                     public void operationComplete(Future<Object> future) throws Exception {
-                        resolvers.remove(executor);
+                        synchronized (resolvers) {
+                            resolvers.remove(executor);
+                        }
                         newResolver.close();
                     }
                 });
